### PR TITLE
chore: call out requirement for allowing commits to your pr branch in contributing guide

### PR DESCRIPTION
**Pull request by kaizencc**
The new markdown syntax is from [here](https://github.com/orgs/community/discussions/16925)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -487,7 +487,15 @@ CDK integration tests.
 
 * Create a commit with your changes and push them to a
   [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
-  > Note: CDK core members can push to a branch on the AWS CDK repo (naming convention: `<user>/<feature-bug-name>`).
+  > **Important**
+  > We will not be able to accept your contribution if you do not allow commits to your PR branch.
```
Comment (kaizencc) [inline CONTRIBUTING.md:491]:
```suggestion
  > We will not be able to accept your contribution if you do not allow commits to your PR branch, as it introduces
  > friction in our review process and breaks our automation that syncs with main. In these scenarios, we will close
  > your pull request and ask that you recreate it with the necessary permissions.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -487,7 +487,17 @@ CDK integration tests.
 
 * Create a commit with your changes and push them to a
   [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
-  > Note: CDK core members can push to a branch on the AWS CDK repo (naming convention: `<user>/<feature-bug-name>`).
+  > **Important**
+  > We will not be able to accept your contribution if you do not allow commits to your PR branch, as it introduces
+  > friction in our review process and breaks our automation that syncs with main. In these scenarios, we will close
```
Comment (kaizencc) [inline CONTRIBUTING.md:492]:
```suggestion
  > friction in our review process and breaks our automation that syncs with the main branch. In these scenarios, we will close
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -487,7 +487,17 @@ CDK integration tests.
 
 * Create a commit with your changes and push them to a
   [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
-  > Note: CDK core members can push to a branch on the AWS CDK repo (naming convention: `<user>/<feature-bug-name>`).
+  > **Important**
```
Comment (kaizencc) [inline CONTRIBUTING.md:490]:
```suggestion
  > [!IMPORTANT]
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -487,7 +487,17 @@ CDK integration tests.
...
+  > We will not be able to accept your contribution if you do not allow commits to your PR branch, as it introduces
+  > friction in our review process and breaks our automation that syncs with the main branch. In these scenarios, we will close
+  > your pull request and ask that you recreate it with the necessary permissions.
+  > This means that you must contribute from a fork within your personal account and also develop
+  > your contribution on a branch other than `main`. See
+  > [Allowing changes to a pull request branch created from a fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
+  > for more information.
+  
+  > **Note**
```
Comment (kaizencc) [inline CONTRIBUTING.md:499]:
```suggestion
  > [!NOTE]
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -593,9 +603,10 @@ dependency.**
 
 #### Adding new unconventional dependencies
 
-> :warning: Do not add these. If there is a tool that you want to use in your
-CDK constructs, see [Adding construct runtime
-dependencies](#Adding-construct-runtime-dependencies).
+> **Warning**
```
Comment (kaizencc) [inline CONTRIBUTING.md:606]:
```suggestion
> [!WARNING]
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -487,7 +487,17 @@ CDK integration tests.
 
 * Create a commit with your changes and push them to a
   [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
-  > Note: CDK core members can push to a branch on the AWS CDK repo (naming convention: `<user>/<feature-bug-name>`).
+  > **Important**
+  > We will not be able to accept your contribution if you do not allow commits to your PR branch, as it introduces
+  > friction in our review process and breaks our automation that syncs with the main branch. In these scenarios, we will close
+  > your pull request and ask that you recreate it with the necessary permissions.
+  > This means that you must contribute from a fork within your personal account and also develop
```
[the remark this answers is not in the document]
Comment (kaizencc) [reply CONTRIBUTING.md:494]: organiZation

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -487,7 +487,17 @@ CDK integration tests.
 
 * Create a commit with your changes and push them to a
   [fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo).
-  > Note: CDK core members can push to a branch on the AWS CDK repo (naming convention: `<user>/<feature-bug-name>`).
+  > [!IMPORTANT]
+  > We will not be able to accept your contribution if you do not allow commits to your PR branch, as it introduces
+  > friction in our review process and breaks our automation that syncs with the main branch. In these scenarios, we will close
+  > your pull request and ask that you recreate it with the necessary permissions.
+  > This means that you must contribute from a fork within your personal account (as opposed to an organisation owned account) and also develop
```
Comment (mrgrain) [inline CONTRIBUTING.md:494]:
```suggestion
  > This means that you must contribute from a fork within your personal account (as opposed to an organization owned account) and also develop
```

**Review by mrgrain — APPROVED** [review]

File: CONTRIBUTING.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mrgrain) [reply CONTRIBUTING.md]: Thanks, but why no fixy? 🥹

**Pull request merged**